src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbol.cs

```diff
@@ -175,7 +174,10 @@ public override OneOrMany<SyntaxList<AttributeListSyntax>> GetAttributeDeclarati
         {
             // Attributes on partial properties are owned by the definition part.
             // If this symbol has a non-null PartialDefinitionPart, we should have accessed this method through that definition symbol instead
-            Debug.Assert(PartialDefinitionPart is null);
+            Debug.Assert(PartialDefinitionPart is null
+                // We might still get here when asking for the attributes on a backing field.
+                // This is an error scenario (requires using a property initializer and field-targeted attributes on partial property implementation part).
+                || this.BackingField is not null);
```

Comment:
@cston I was anticipating that perhaps the synthesized BackingField symbol will be shared between partial property parts, or otherwise the two field symbols will reference each other similarly as the associated partial property symbols. But, this work won't happen until field keyword feature is implemented, and how specifically it will be done is TBD.